integrations/nginx/install.go

```diff
@@ -0,0 +1,349 @@
+	}
+
+	uninstalledError = &InstallationError{
+		Wrapped: errors.New("NGINX is not installed, or 'nginx' is not in the current path"),
+		Remedy:  "Please run this command on the machine where NGINX is installed.",
+	}
+
+	downloadError = &InstallationError{
+		Wrapped: errors.New("The precompiled module could not be downloaded from Akita."),
+		Remedy:  "Please download the module from https://github.com/akitasoftware/akita-nginx-module/releases/latest and complete the installation by copying it into the NGINX module directory.",
+	}
+
+	symlinkError = &InstallationError{
+		Wrapped: errors.New("The precompiled module was installed, but a symbolic link to it could not be created."),
+		Remedy:  "Please create a symbolic link in the module directory from ngx_http_akita_module.so to the installed module.",
+	}
+)
+
+func NewCopyError(path string) error {
+	return &InstallationError{
+		Wrapped: errors.New("The precompiled module could not be installed into the NGINX directory."),
+		Remedy:  fmt.Sprintf("Please complete the installation by copying %s into the NGINX module directory.\n", path),
+	}
+}
+
+type InstallArgs struct {
+	DryRun bool
+
+	// Destination directory for the module
+	DestDir optionals.Optional[string]
+}
+
+// Install the precompiled NGINX module from the latest release on GitHub.
+// If we can't find the Nginx version or platform,
+// Return an error to show to the user.
+func InstallModule(args *InstallArgs) error {
+	version, err := FindNginxVersion()
+	if err != nil {
+		telemetry.Error("NGINX find version", err)
+		return err
+	}
+
+	arch, platform, err := FindPlatform()
+	if err != nil {
+		telemetry.Error("NGINX find platform", err)
+		return err
+	}
+
+	// Report the version being attempted
+	telemetry.InstallIntegrationVersion("NGINX", arch, platform, version)
+
+	// The downside of using Github is we don't get any hierarchy, so it
+	// all has to be crammed into the name.
+	expectedFilename := fmt.Sprintf("ngx_http_akita_module_%s_%s_%s.so",
+		arch, platform, version)
+	printer.Debugf("Looking for release artifact %q\n", expectedFilename)
+
+	// Connect to Github and grab the latest release.
+	assets, err := GetLatestReleaseAssets()
+	if err != nil {
+		telemetry.Error("NGINX get release", err)
+		return downloadError
+	}
+
+	// Find the matching prebuilt modules
+	var assetOpt optionals.Optional[GithubAsset]
+	for _, a := range assets {
+		if a.Name == expectedFilename {
+			assetOpt = optionals.Some(a)
+		}
+	}
+	asset, present := assetOpt.Get()
+	if !present {
+		return unsupportedError
+	}
+
+	printer.Infof("Selected prebuilt module %s\n", asset.Name)
+
+	// Find the Nginx directory
+	destDir := args.DestDir
+	if destDir.IsNone() {
+		destDir = FindNginxModuleDir()
+	}
+	if destDir.IsNone() {
+		printer.Warningf("Can't identify directory for module to be installed.\n")
+	} else {
+		printer.Infof("Module will be installed in %v\n", destDir.GetOrDefault(""))
+	}
```

Comment:
Avoid unnecessary `GetOrDefault`.
```suggestion
	if dest, exists := destDir.Get(); exists {
		printer.Infof("Module will be installed in %v\n", dest)
	} else {
		printer.Warningf("Can't identify directory for module to be installed.\n")
	}
```